src/python/AsyncStageOut/TransferWorker.py

```diff
@@ -307,13 +341,30 @@ def command(self, jobs, jobs_lfn, jobs_pfn, jobs_report):
             # Validate copyjob file before doing anything
             self.logger.debug("Valid %s" % self.validate_copyjob(copyjob))
             if not self.validate_copyjob(copyjob): continue
-            rest_copyjob = '{"params":{"bring_online":null,"verify_checksum":false,"reuse":false,"copy_pin_lifetime":-1,"max_time_in_queue":20,"job_metadata":{"issuer": "ASO"},"spacetoken":null,"source_spacetoken":null,"fail_nearline":false,"overwrite":true,"gridftp":null},"files":['
+            rest_copyjob = {"params":
+                                    {"bring_online": None,
+                                     "verify_checksum": False,
+                                     "reuse": reuse,
+                                     "copy_pin_lifetime": -1,
+                                     "max_time_in_queue": 20,
+                                     "job_metadata":
+                                                    {"issuer": "ASO"},
+                                     "spacetoken": None,
+                                     "source_spacetoken": None,
+                                     "fail_nearline": False,
+                                     "overwrite": True,
+                                     "gridftp": None},
+                            "files":[]
+                           }
             pairs = []
             for SrcDest in copyjob:
-                pairs.append('{"sources":["' + SrcDest.split(" ")[0] + '"],"metadata":null,"destinations":["' + SrcDest.split(" ")[1] + '"]}')
-            rest_copyjob += (','.join(pairs)) +']}'
-            self.logger.debug("Subbmitting this REST copyjob %s" % rest_copyjob)
-            post = urllib.quote(rest_copyjob)
+                tempDict = {"sources": [], "metadata": None, "destinations": []}
+                tempDict["sources"].append(SrcDest.split(" ")[0])
+                tempDict["destinations"].append(SrcDest.split(" ")[1])
+                rest_copyjob.append(tempDict)
+
```

Comment:
Pls dump the request into json before submitting it to FTS. FTS team was always compaining that the requests were not valid json. Maybe we can take this opportunity to fix it.